crates/bevy_reflect/src/reflect.rs

```diff
@@ -11,6 +11,22 @@ use std::{
 use crate::utility::NonGenericTypeInfoCell;
 pub use bevy_utils::AHasher as ReflectHasher;
 
+/// A simple enumeration of "kinds" of type, without any associated object.
+///
+/// A `ReflectKind` is obtained via [`Reflect::reflect_kind`].
+#[derive(Copy, Clone, Debug)]
```

Comment:
Could we also derive `PartialEq` and `Eq` here? I can see users wanting to do `a == b` rather than pattern matching all possible cases.

I don't think `Hash` is super necessary, but do you think we should add that as well?

Reply:
I guess we can, I assume it will get optimized out if one does not use it.